### DISCUSSION TITLE
Can I change the default locale and locales in my i18n.js config dynamically?

### DISCUSSION THREAD

**Discussion by BozhidarPeychevOSF** [vinissimus/next-translate#839] [Q&A]
Hey guys, currently I'm facing a problem that would be resolved simply by being able to change the config dynamically. This is due to the fact that in our project we have multiple domains with different allowed locales, locales and default locales. Since in some of them the default locale is the same I cannot rely on the domains property inside the config or in the next.config.js config.

My question is if there is an option to change the locales and the defaultLocale properties dynamically inside of the config files or whether you might have any suggestions to handle the aforementioned case?

**Comment by aralroca** [comment]
what is the use case?  This configuration is part of the Next.js core: https://nextjs.org/docs/advanced-features/i18n-routing, maybe you can use `domains`:

```js
domains: [
      {
        domain: 'example.com',
        defaultLocale: 'en-US',
      },
      {
        domain: 'example.nl',
        defaultLocale: 'nl-NL',
      },
      {
        domain: 'example.fr',
        defaultLocale: 'fr',
        // an optional http field can also be used to test
        // locale domains locally with http instead of https
        http: true,
      },
    ],
```

**Reply by BozhidarPeychevOSF** [reply]
Hello Aral, 
I cannot use domains since the Next.js core has a limit that doesn't allow the default locale to be duplicated for different domains.
In this example if we have "en" set as a default locale more than once the config will crash with an error saying you cannot use the same default locale more than once despite having different locales set.

domains: [
      {
        domain: 'example.com',
        defaultLocale: 'en',
        locales: ['en', 'es'],
      },
      {
        domain: 'example.nl',
        defaultLocale: 'en',
        locales: ['en', 'nl'],
      },
      {
        domain: 'example.fr',
        defaultLocale: 'fr',
        locales: ['en', 'fr'],
        // an optional http field can also be used to test
        // locale domains locally with http instead of https
        http: true,
      },
    ],

With this said the use case is an option to have the same `defaultLocale` for different domains while having different locales available if we use the in build domains option in the config. I just wanted to also check whether it was possible to make the config dynamic or if there is an option to do what I stated above.
